From e6435969a0ef256049da9c3cf4a7428938b8d93f Mon Sep 17 00:00:00 2001
From: Cyril Chemparathy <cyril.chemparathy@xilinx.com>
Date: Thu, 7 Sep 2017 14:45:15 -0700
Subject: [PATCH] libxc: avoid uninitialized use domctl field

This patch fixes the following warning/error:

xc_dom_arm.c: In function 'meminit':
xc_dom_arm.c:229:31: error: 'domctl.u.address_size.size' may be used uninitialized in this function [-Werror=maybe-uninitialized
     if ( domctl.u.address_size.size == 0 )
     ~~~~~~~~~~~~~~~~~~~~~^~~~~
---
 tools/libxc/xc_dom_arm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/libxc/xc_dom_arm.c b/tools/libxc/xc_dom_arm.c
index e7d4bd0..bac5e75 100644
--- a/tools/libxc/xc_dom_arm.c
+++ b/tools/libxc/xc_dom_arm.c
@@ -223,6 +223,7 @@ static int set_mode(xc_interface *xch, domid_t domid, char *guest_type)
 
     domctl.domain = domid;
     domctl.cmd    = XEN_DOMCTL_set_address_size;
+    domctl.u.address_size.size = 0;
     for ( i = 0; i < ARRAY_SIZE(types); i++ )
         if ( !strcmp(types[i].guest, guest_type) )
             domctl.u.address_size.size = types[i].size;
-- 
2.7.4

